### PR TITLE
C# 9 reference update: target-typed new

### LINKED ISSUE #19705: Docs for What's new in C# 9.0
C# 9 feature status: https://github.com/dotnet/roslyn/blob/master/docs/Language%20Feature%20Status.md

@BillWagner I think one year ago we've worked quite efficiently for C# 8.0 doc updates:

- You've created the main page for "What's new in C# 8.0" with the major new features described
- I've updated that page and the related reference doc with smaller new features

Would it work this year for C# 9.0? I'm thinking of updating the docs for `new` (target-typed new), the conditional operator (target-typed conditional), `foreach` (extension `GetEnumerator`), lambda expressions (discard parameters).

### LINKED ISSUE #20580: Language reference: target typed new
- Update `new` page for target typed new.
- Update `var` page to describe target type `new` as an alternative.

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #20580 
Fixes #19705 (there are better issues that cover the same area)

### CODE REVIEW

File: docs/csharp/language-reference/keywords/var.md [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
+
+```csharp
+var xs = new List<int>();
+```
+
+Beginning with C# 9.0, you can use a target-typed [`new` expression](../operators/new-operator.md) as an alternative:
+
+```csharp
+List<int> xs = new();
+```
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/var.md:34]: To contrast with `var` and the *important* note, I'd recommend adding a second example with a nullable type: 

```suggestion
```csharp
List<int> xs = new(); // non-nullable.
List<int>? xy = new(); // nullable.
```
```

**Review by BillWagner — COMMENTED** [review]
Commenting so it stays in my queue.

This is great @pkulikov 

I had one suggestion, and I'd your thoughts on it.

Once we resolve that discussion, this is ready to :shipit:

File: docs/csharp/language-reference/keywords/var.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/keywords/var.md:34]: @BillWagner I would remove the comments, though. What do you think?

Comment (BillWagner) [reply docs/csharp/language-reference/keywords/var.md:34]: Yes. I updated the suggestion.

Comment (pkulikov) [reply docs/csharp/language-reference/keywords/var.md]: Thank you, I've applied the suggestion

File: docs/csharp/language-reference/keywords/var.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Beginning with C# 9.0, you can use a target-typed [`new` expression](../operator
 
 ```csharp
 List<int> xs = new();
-List<int>? xy = new();
+List<int>? ys = new();
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/var.md:34]: This is likely my mistake, but the closing backs ticks are missing:

Comment (pkulikov) [reply docs/csharp/language-reference/keywords/var.md:34]: Thank you for the catch @BillWagner. They are back.

**Review by BillWagner — APPROVED** [review]
This is great @pkulikov 

I'll :shipit: now.


Thanks again for all your help!

**Pull request merged**